src/App.js

```diff
@@ -0,0 +1,69 @@
+import "./App.css";
+import React from "react";
+import peopleData from "./peopleData";
+import SearchField from "./SearchField";
+import CreateTable from "./CreateTable";
+import { prepareData, isSpecial } from "./myMethods";
+
+let people = prepareData(peopleData);
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      people: people,
+      direction: "asc"
+    };
+
+    this.sortBy = this.sortBy.bind(this);
+    this.filterList = this.filterList.bind(this);
+  }
+
+  filterList(event) {
+    let value = event.target.value.toLowerCase();
+    let filteredPeople = this.state.people.filter(person => {
+      for (let key in person) {
+        let keyValue = person[key].toString().toLowerCase();
+
+        if (keyValue.includes(value)) {
+          return true;
+        }
+      }
+    });
+
+    this.setState({
+      people: filteredPeople.length > 0 ? filteredPeople : people
+    });
+  }
+
+  sortBy(key) {
+    this.setState({
+      people: people.sort((a, b) => {
```

Comment:
I'd recommend create separate methods `sortStrings` and `sortNumbers`

Reply:
Also, you mutate state with sorting `people` directly.
```suggestion
      people: [...people].sort((a, b) => {
```